crates/sui/src/client_commands.rs

```diff
@@ -866,6 +867,8 @@ impl SuiClientCommands {
                 let sender = sender.unwrap_or(context.active_address()?);
                 let client = context.get_client().await?;
                 let chain_id = client.read_api().get_chain_identifier().await.ok();
+                let protocol_config =
+                    ProtocolConfig::get_for_version(ProtocolVersion::MAX, Chain::Unknown);
```

Comment:
You should be able to convert the `chain_id` from the line about into a `Chain` enum.

Reply:
I think we might need this function. I'm having trouble finding it. searching the project for '35834a8a' mainnet id has some references but none lead me to a method of converting to the enum.

Reply:
I'm unsure about testnet since it gets wiped but maybe this?
https://github.com/jordanjennings-mysten/sui/commit/4cc83ef2438091768e2f5e2e982c88b57ff6d31d

Reply:
This is what I was thinking of:

https://github.com/MystenLabs/sui/blob/54ccf09ef3764b5b94082860bcba858a15ce6ab7/crates/sui-types/src/digests.rs#L182

But it's a little annoying to use in this context, because you are given the chain identifier as a string, and that string only contains the first 4 bytes of the checkpoint digest. I think your solution (adding a new function for this) seems reasonable, but I would:

- Rely on the digests file to act as a source of truth for what the chain identifiers look like for mainnet and testnet so that if testnet is wiped, there's only a single place that needs to be updated.
- Add docs explaining why this function exists (really the issue is that the RPC Chain Identifier is contracted.

Reply:
I think this is resolved, shifted some things around. I think my change to digests makes the most sense given that `sui-protocol-config` is a dependency of `sui types`. In other words I cannot import the chain identifier b58 into the module where `Chain` exists because of a dependency loop so I pulled up the logic into sui types to match short hex ids.